cmake/EkatMpiUtils.cmake

```diff
@@ -1,28 +1,36 @@
 # Detect the library that provides MPI
 set (EKAT_CMAKE_DIR ${CMAKE_CURRENT_LIST_DIR})
 macro (GetMpiDistributionName DISTRO_NAME)
-  if (NOT MPIEXEC_EXECUTABLE)
+  if (MPI_C_COMPILER)
+    set (COMPILER_NAME ${MPI_C_COMPILER})
+  elseif (MPI_CXX_COMPILER)
+    set (COMPILER_NAME ${MPI_CXX_COMPILER})
+  else ()
     string (CONCAT MSG
-            "MPIEXEC_EXECUTABLE is not set. You must call find_package(MPI ..),\n"
-            "*before* calling GetMpiDistributionName (and in the same scope)")
+      "MPI_C_COMPILER and MPI_CXX_COMPILER are not set.\b"
+      "CMake logic to determine the distribution name requires C/CXX mpi compilers.\n"
+      "Please call find_package(MPI COMPONENTS [C|CXX]) *before* calling GetMpiDistributionName (and in the same scope)")
     message ("${MSG}")
     message (FATAL_ERROR "Aborting")
   endif()
-  execute_process(COMMAND ${MPIEXEC_EXECUTABLE} --version
+  execute_process(COMMAND ${COMPILER_NAME} -c ${EKAT_CMAKE_DIR}/TryCompileMPI.cxx
```

Comment:
Have you ever used CMake's [try_compile](https://cmake.org/cmake/help/latest/command/try_compile.html) command? Just curious. It seems to be intended for this stuff.

Reply:
Doesn't try-compile use `CMAKE_<LANG>_COMPILER` to compile? Right now, it is technically possible to build EKAT using standard compilers, and link MPI to them. Here, however, I want to use the MPI compiler wrappers. I guess I could tell cmake to include mpi when running this specific try compile...

I might switch to try_compile (with MPI include flags), since I suspect that that's more portable. Manually running commands always has the chance of causing shady config errors down the road.